App/src/main/java/cc/softwarefactory/lokki/android/utilities/Utils.java

```diff
@@ -386,4 +387,17 @@ private static String getInitials(String text) {
         return result;
     }
 
+    public static void setVisibility (boolean visible, Context context) {
+        try {
+            MainApplication.visible = visible;
+            ServerApi.setVisibility(context, visible);
+            if (visible == false) {
+                LocationService.stop(context);
+            } else {
+                LocationService.start(context);
+            }
+        } catch (Exception ex) {
```

Comment:
You should catch the specific exception that the try block is throwing rather than catching all exceptions.